api/v1beta1/machinedeployment_webhook.go

```diff
@@ -76,6 +77,18 @@ func (m *MachineDeployment) ValidateDelete() error {
 
 func (m *MachineDeployment) validate(old *MachineDeployment) error {
 	var allErrs field.ErrorList
+	// The MachineDeployment name is used as a label value. This check ensures that longer names which could not be
+	// valid label values are rejected.
+	if len(m.Name) > validation.LabelValueMaxLength {
```

Comment:
should we use IsValidLabelValue instead, so we validate the name can be a valid label (which is the end goal)

Reply:
same for cluster web hook